### ISSUE TITLE
webpack throws when node executable is not in PATH

### ISSUE THREAD

**Issue by guest271314** [oven-sh/bun#7479]
### What version of Bun is running?

1.0.15+b3bdf22eb

### What platform is your computer?

Linux 5.15.0-43-generic x86_64 x86_64

### What steps can reproduce the bug?

Make sure `node` executable is not in `PATH`.

```
$ git clone https://github.com/guest271314/webbundle
$ cd webbundle
$ bun install
$ bun run build
```

### What is the expected behavior?

No errors to be thrown

### What do you see instead?

```
$ bun run build
$ webpack --config webpack.wbn.js
[bun] Warning: worker_threads.Worker option "eval" is not implemented.
[bun] Warning: worker_threads.Worker option "stdout" is not implemented.
[bun] Warning: worker_threads.Worker option "stderr" is not implemented.
[bun] Warning: worker_threads.Worker option "resourceLimits" is not implemented.
error: [native code]
error: script "build" exited with code 1 (SIGHUP)
```

### Additional information

After I fetch the `node` executable and place in `PATH` no errors are thrown.

**Comment by paperclover** [comment]
two issues:

- likely need to implement those worker options. see #4165
- horrible error printing on our part, not sure

**Comment by guest271314** [comment]
Why is the error only thrown when `node` executable is not in `PATH`? 

Is `bun` calling `node` executable in some cases?

**Comment by paperclover** [comment]
the `webpack` cli defines `#!/usr/bin/env node`, which bun respects. there is a fallback for if node is not installed.

**Comment by guest271314** [comment]
`bun` throws when calling `webpack()` directly.

```
const webpack = require("webpack");
const WebBundlePlugin = require("webbundle-webpack-plugin");
const { WebBundleId, parsePemKey } = require("wbn-sign");
const { merge } = require("webpack-merge");
const HtmlWebpackPlugin = require("html-webpack-plugin");
const CopyPlugin = require("copy-webpack-plugin");
const fs = require("node:fs");
const path = require("node:path");

const privateKeyFile = "ed25519key.pem";
const privateKey = fs.readFileSync(privateKeyFile);
const parsedPrivateKey = parsePemKey(privateKey);
const baseURL = new WebBundleId(
  parsedPrivateKey,
).serializeWithIsolatedWebAppOrigin();
console.log(baseURL);
const webBundlePlugin = new WebBundlePlugin({
  baseURL,
  output: "signed.swbn",
  integrityBlockSign: {
    key: parsedPrivateKey,
  },
});

webpack({
  entry: "./src/script.js",
  module: {
    rules: [
      {
        test: /\.html$/,
        loader: "html-loader",
      },
    ],
  },
  mode: "production",
  plugins: [
    webBundlePlugin,
    new HtmlWebpackPlugin({
      template: "src/index.html",
    }),
    new CopyPlugin({
      patterns: [
        { from: "assets" },
      ],
    }),
  ],
  resolve: {
    extensions: [".js"],
  },
  output: {
    path: path.resolve(__dirname, "dist"),
    publicPath: "./",
    trustedTypes: {
      policyName: "webbundle#webpack",
    },
  },
}, (err, stats) => {
  if (err || stats.hasErrors()) {
    console.log(err);
  }
  console.log(stats.compilation.compiler.outputPath, 
    stats.compilation.assets,
    [...stats.compilation.modules][0].request
  );
});
```

```
$ node webpack.wbn.js
isolated-app://<ID>/
Setting the empty headerOverrides to IWA defaults. To bundle a non-IWA, set `integrityBlockSign { isIwa: false }` in your plugin configs. Defaults are set to:
 {"content-security-policy":"base-uri 'none'; default-src 'self'; object-src 'none'; frame-src 'self' https: blob: data:; connect-src 'self' https:; script-src 'self' 'wasm-unsafe-eval'; img-src 'self' https: blob: data:; media-src 'self' https: blob: data:; font-src 'self' blob: data:; require-trusted-types-for 'script'; frame-ancestors 'self';","cross-origin-embedder-policy":"require-corp","cross-origin-opener-policy":"same-origin","cross-origin-resource-policy":"same-origin"}
{
  'main.js': SizeOnlySource { _size: 57 },
  'manifest.webmanifest': SizeOnlySource { _size: 1303 },
  'index.html': SizeOnlySource { _size: 218 },
  'signed.swbn': SizeOnlySource { _size: 4190 }
}
```
```
bun run webpack.wbn.js
isolated-app://aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaic/
[bun] Warning: worker_threads.Worker option "eval" is not implemented.
[bun] Warning: worker_threads.Worker option "stdout" is not implemented.
[bun] Warning: worker_threads.Worker option "stderr" is not implemented.
[bun] Warning: worker_threads.Worker option "resourceLimits" is not implemented.
error: [native code]
```

**Comment by guest271314** [comment]
@paperdave Evidently `node` implementation of `crypto` module is very Node.js-specific https://www.reddit.com/r/javascript/comments/18pyd4j/askjs_why_is_the_internal_crypto_module_so/
> The crypto module is not entirely Javascript. Some of its implementation uses native code which only runs in the nodejs environment and some of its implementation uses nodejs internals (like the thread pool).

**Comment by guest271314** [comment]
@paperdave I stopped asking qustions and for help and got to work yesterday. I replaced all the `node:crypto` usages with `webcrypto` usages (and substituted `Uint8Array` for `Buffer`) in https://github.com/GoogleChromeLabs/webbundle-plugins/tree/main/packages/rollup-plugin-webbundle. 

The same code now runs using `bun`, `deno`, and `node` without errors and produces a Signed Web Bundle for an Isolated Web App. I found `esbuild` and Rollup version simpler to work with.

I'm not sure if `node:crypto` can be implemented outside of `node`.
